docs/contributing/docs-and-website-components.md

```diff
@@ -0,0 +1,409 @@
+
+It can be used like this:
+
+```markdown:title=docs/using-gatsby-image
+---
+title: Using Gatsby Image
+---
+
+This is how you use `gatsby-image`...
+
+<!-- highlight-start -->
+
+<EggheadEmbed
+  lessonLink="https://egghead.io/lessons/gatsby-use-gatsby-image-s-graphql-fragments-for-blurred-up-and-traced-svg-images"
+  lessonTitle="Use gatsby-image's GraphQL fragments for blurred-up and traced SVG images"
+/>
+
+<!-- highlight-end -->
+```
+
+#### Sample
+
+Rendered, it looks like this:
+
+<EggheadEmbed
+  lessonLink="https://egghead.io/lessons/gatsby-use-gatsby-image-s-graphql-fragments-for-blurred-up-and-traced-svg-images"
+  lessonTitle="Use gatsby-image's GraphQL fragments for blurred-up and traced SVG images"
+/>
+
+---
+
+### Pull quote
+
+The `<Pullquote />` component is used to call out a quote in the blog. It applies borders and styles that make a section of the content pop out to readers.
+
+#### Usage
+
+It takes one prop, and uses the children it wraps to populate its inner content:
+
+- `citation` - the reference of the person or entity that made the quoted statement
+
+It is used like this:
+
+```markdown:title=blog/new-announcement.md
+---
+title: Big News!
+author: Jay Gatsby
+---
+
+<!-- highlight-start -->
+<Pullquote citation="Winston Churchill">
+  To improve is to change, so to be perfect is to have changed often.
+</Pullquote>
+<!-- highlight-end -->
+```
+
+#### Sample
+
+Rendered, the component looks like this:
+
+<Pullquote citation="Winston Churchill">
+  To improve is to change, so to be perfect is to have changed often.
+</Pullquote>
+
+---
+
+## Other available components
+
+Other less commonly used components aren't globally available, but can imported at the top of a markdown file and used in other docs.
+
+---
+
+### Layer model
+
+The `<LayerModel />` was made to help explain concepts of how Gatsby works at a high level. It conceptually breaks Gatsby into different layers and shows how data is pulled, aggregated, and eventually rendered as an app. It's used on pages to help explain and visual how Gatsby works at different levels.
+
+#### Usage
+
+The component takes one prop:
+
+- `initialLayer` - defaults to `Content`, can be one of `Content`, `Build`, `Data`, `View`, `App` that correspond to the different layers
+
+```markdown
+---
+title: GraphQL Concepts in Gatsby
+---
+
+import LayerModel from "../../www/src/components/layer-model"
+
+To help understand how GraphQL works in Gatsby...
+
+<LayerModel initialLayer="Data" />
+```
+
+#### Sample
+
+When used, it looks like this:
+
+<LayerModel initialLayer="Data" />
+
+### Horizontal navigation list
+
+The `<HorizontalNavList />` was made for the [Glossary](/docs/glossary/), and renders a list of links to subheadings on the page in a horizontal format. Because of it's specific use case, it isn't made globally available and can be imported if it's needed on other pages.
```

Comment:
```suggestion
The `<HorizontalNavList />` was made for the [Glossary](/docs/glossary/), and renders a list of links to alphabetical subheadings on the page in a horizontal format. Because of its specific use case, it isn't made globally available but can be imported if needed on other pages.
```